### PR TITLE
Fix edge detection and reenable entropy crop tests

### PR DESCRIPTION
**Pull request by JimBobSquarePants**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/SixLabors/ImageSharp/pulls) open
- [x] I have verified that I am following matches the existing coding patterns and practice as demonstrated in the repository. These follow strict Stylecop rules :cop:.
- [x] I have provided test coverage for my change (where applicable)

### Description
Edge detection now produces fully opaque output. This stabilizes the output across platforms and also fixes entropy cropping.

### CODE REVIEW

**Comment by antonfirsov** [comment]
I tried to follow the discussions under #1143, but maybe missed a few points. What is the main reason you've chosen this vs premultiplication?

I wonder how does [OpenCV](https://docs.opencv.org/trunk/da/d22/tutorial_py_canny.html) or [ImageMagic](https://imagemagick.org/discourse-server/viewtopic.php?t=25405) work for images with alpha. @dlemstra do you know or can you quickly check it?

File: src/ImageSharp/Processing/Processors/Filters/OpaqueProcessor{TPixel}.cs [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+                PixelOperations<TPixel>.Instance.ToVector4(this.configuration, targetRowSpan.Slice(0, span.Length), span, PixelConversionModifiers.Scale);
+                ref Vector4 baseRef = ref MemoryMarshal.GetReference(span);
+
+                for (int x = 0; x < this.bounds.Width; x++)
+                {
+                    ref Vector4 v = ref Unsafe.Add(ref baseRef, x);
+                    v.W = 1F;
+                }
+
+                PixelOperations<TPixel>.Instance.FromVector4Destructive(this.configuration, span, targetRowSpan, PixelConversionModifiers.Scale);
```
Comment (antonfirsov) [inline src/ImageSharp/Processing/Processors/Filters/OpaqueProcessor{TPixel}.cs]: Yet another place where an extended `PixelTypeInfo` would be super useful to eliminate unnecessary scanning. [Related discussion](https://github.com/saucecontrol/PhotoSauce/pull/39#discussion_r392411909).

Comment (JimBobSquarePants) [reply src/ImageSharp/Processing/Processors/Filters/OpaqueProcessor{TPixel}.cs]: Yup.

**Comment by JimBobSquarePants** [comment]
@antonfirsov We do use premultiplication but then reassign the alpha value from the source before unpremultiplying. If you don't do that you end up with a blank image if you have a consistent alpha channel. 

Basically the only thing I changed was to normalize the alpha component before the algorithm runs. 

OpenCV does the [same](https://stackoverflow.com/questions/49071592/opencv-edge-detection-on-png-image-with-alpha-layer). In the link you can see it's picking up changes that would not exist without removing the alpha component first.

PIL doesn't do this and will produce an [entirely transparent image](https://github.com/python-pillow/Pillow/issues/4321). (However in that closed issue they've misidentified the result as having a zero alpha channel, when, in fact, the image has no information in any channels.

File: src/ImageSharp/Processing/Processors/Filters/OpaqueProcessor{TPixel}.cs [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+using System;
+using System.Numerics;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using SixLabors.ImageSharp.Advanced;
+using SixLabors.ImageSharp.PixelFormats;
+
+namespace SixLabors.ImageSharp.Processing.Processors.Filters
+{
+    internal sealed class OpaqueProcessor<TPixel> : ImageProcessor<TPixel>
```
Comment (antonfirsov) [inline src/ImageSharp/Processing/Processors/Filters/OpaqueProcessor{TPixel}.cs]: Wouldn't be `MakeOpaqueProcessor` better? (Related extension to be added later: `image.MakeOpaque()`.)

Comment (JimBobSquarePants) [reply src/ImageSharp/Processing/Processors/Filters/OpaqueProcessor{TPixel}.cs]: We can do that if/when we choose to expose such an extension.

**Pull request merged**